tools/api_proto_plugin/annotations.py

```diff
@@ -63,5 +67,43 @@ def ExtractAnnotations(s, inherited_annotations=None):
   return annotations
 
 
+def XformAnnotation(s, annotation_xforms):
+  """Return transformed string with annotation transformers.
+
+  The annotation will be replaced with the new value returned by the transformer.
+  If the transformer returns None, then the annotation will be removed.
+  If the annotation presented in transformers doesn't exist in the original string,
+  a new annotation will be appended to the end of string.
+
+  Args:
+    annotation_xforms: a dict of transformers for annotations.
+
+  Returns:
+    transformed string.
+  """
+  present_annotations = set()
+
+  def xform(match):
+    annotation, content, trailing = match.groups()
+    present_annotations.add(annotation)
+    annotation_xform = annotation_xforms.get(annotation)
+    if annotation_xform:
+      value = annotation_xform(annotation)
+      return "[#%s:%s]%s" % (annotation, value, trailing) if value is not None else ""
+    else:
+      return match.group(0)
+
+  def append(s, annotation, content):
+    return "%s [#%s:%s]\n" % (s, annotation, content)
```

Comment:
Nit: for consistency, switch all the `"style strings"` to `'style strings'`, that's the standard in these files.